homeassistant/components/sensor/repetier.py

```diff
@@ -0,0 +1,223 @@
+                'done':
+                    self._data.data[self._repetier_id]['done'],
+                'job':
+                    self._data.data[self._repetier_id]['job'],
+                'jobid':
+                    self._data.data[self._repetier_id]['jobid'],
+                'linesSend':
+                    self._data.data[self._repetier_id]['linesSend'],
+                'ofLayer':
+                    self._data.data[self._repetier_id]['ofLayer'],
+                'pauseState':
+                    self._data.data[self._repetier_id]['pauseState'],
+                'paused':
+                    self._data.data[self._repetier_id]['paused'],
+                'printTime':
+                    self._data.data[self._repetier_id]['printTime'],
+                'printedTimeComp':
+                    self._data.data[self._repetier_id]['printedTimeComp'],
+                'start':
+                    self._data.data[self._repetier_id]['start'],
+                'totalLines':
+                    self._data.data[self._repetier_id]['totalLines'],
+                'online':
+                    self._data.data[self._repetier_id]['online'],
+            }
+        else:
+            if SHOW_PCT:
+                self._state = STATE_IDLE
+            else:
+                self._state = STATE_IDLE
+            self._attributes = {
+                'active':
+                    self._data.data[self._repetier_id]['active'],
+                'job':
+                    self._data.data[self._repetier_id]['job'],
+                'pauseState':
+                    self._data.data[self._repetier_id]['pauseState'],
+                'paused':
+                    self._data.data[self._repetier_id]['paused'],
+                'online':
+                    self._data.data[self._repetier_id]['online'],
+            }
+            self._units = None
+    else:
+        self._state = STATE_OFF
+        self._attributes = {
+            'active':
+                self._data.data[self._repetier_id]['active'],
+            'job':
+                self._data.data[self._repetier_id]['job'],
+            'pauseState':
+                self._data.data[self._repetier_id]['pauseState'],
+            'paused':
+                self._data.data[self._repetier_id]['paused'],
+            'online':
+                self._data.data[self._repetier_id]['online'],
+        }
+        self._units = None
+
+
+class RepetierData():
+    """Get the latest sensor data."""
+
+    def __init__(self, parser, config):
+        """Initialize the data object."""
+        url = config.get(CONF_URL)
+        port = config.get(CONF_PORT)
+        api_key = config.get(CONF_API_KEY)
+        self.url = url + ':' + port + '/printer/list/?apikey=' + api_key
+        self.data = None
+        self.repetier_api_response = parser
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Get the latest data."""
+        response = requests.get(self.url)
```

Comment:
The service specific code, I/O and data parsing, should be extracted into a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

Reply:
Thank you - that's way over my skill level for Python - I'm a C#/PHP kinda guy ;)
Will close this request and perhaps - if time will allow - open a new request if and when I have had the opportunity to lean Pyhton a bit more in depth.

Reply:
I'm trying to understand what you want, taking the docs into account.
So just to make sure, @MartinHjelmare, you want the RepetierData class in a PyPi module rather than in this sensor component?
The docs seems a bit uncomplete to me on that point, regarding how much we are allowed to have in the component and how much you want in an external module.

Reply:
Yes, the `RepetierData` class should be moved to a module or package published on pypi.